src/spac/data_utils.py

```diff
@@ -472,18 +471,18 @@ def select_values(data, annotation, values=None):
     return data
 
 
-def downsample_cells(data, annotation, n_samples=None,
+def downsample_cells(input_data, annotations, n_samples=None,
                      stratify=False, rand=False):
     """
-    Reduces the number of rows in the data by either selecting n_samples from
-    every possible value of annotation(s), or returning n_samples
+    Reduces the number of rows in the input_data by either selecting n_samples
+    from  every possible value of annotations, or returning n_samples
     stratified by the frequency of values in annotations.
 
     Parameters
     ----------
-    data : pd.DataFrame
+    input_data : pd.DataFrame
         The input data frame.
-    annotation : str or list of str
+    annotations : str or list of str
         The column name(s) to downsample on. If multiple column names are
         provided, they are combined into a single column for downsampling.

```

Comment:
@fangliu117 , would you please elaborate how the two columns are combined in one? You mean the unique combinations of values in the two columns is generated?

Reply:
@georgezakinih Thank you for your question. When combining the two columns:
1.	The values of the specified annotation columns for each row are concatenated using an underscore. So if one column has the value "a" and the other "x" for a particular row, the combined column for that row would have the value "a_x".
2.	Yes, the unique combinations of values from the two columns are effectively generated. The combined column represents all possible combinations of values from the original columns. When we group by this combined column later in the code, we inherently work with these unique combinations, which allows us to treat each unique combination as a distinct group. For instance, rows with the same "a_x" combination will be grouped together.

Reply:
Update: to make it clearer, I have updated the docstring with more information in the new commit https://github.com/FNLCR-DMAP/SCSAWorkflow/pull/102/commits/f53cf81de01872afbe0488d7e44588f89aa27b76

Reply:
Thanks @fangliu117, yes that's a clever idea.